subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -354,11 +354,34 @@ struct pdu_adv *lll_adv_pdu_latest_get(struct lll_adv_pdu *pdu,
 		uint8_t pdu_idx;
 		void *p;
 
-		if (!MFIFO_ENQUEUE_IDX_GET(pdu_free, &free_idx)) {
-			return NULL;
-		}
-
 		pdu_idx = first;
+		p = pdu->pdu[pdu_idx];
+
+		do {
+			void *next;
+
+			/* Store partial list in current data index if there is
+			 * no free slot in mfifo. It can be released on next
+			 * switch attempt (on next event).
+			 */
```

Comment:
Currently there will be no next switch attempt because caller will just assert, as it does without linking. So we can assume this is ok as there always should be enough free slots in mfifo or we can add handling in caller later, i.e. it will skip an event.

Alternatively we can put only 1st PDU in mfifo with all linked PDUs and let ULL deal with it, i.e. it will dequeue this PDU and use it, then release linked PDUs directly to pool.